src/librustc/mir/tcx.rs

```diff
@@ -229,9 +229,9 @@ impl<'tcx> Rvalue<'tcx> {
             Rvalue::Repeat(ref operand, count) => {
                 tcx.mk_array(operand.ty(local_decls, tcx), count)
             }
-            Rvalue::Ref(reg, bk, ref place) => {
+            Rvalue::Ref(bk, ref place) => {
                 let place_ty = place.ty(local_decls, tcx).to_ty(tcx);
-                tcx.mk_ref(reg,
+                tcx.mk_ref(tcx.types.re_erased,
```

Comment:
Hmm, I'm not sure how I feel about this. It seems to add some non-trivial complexity that the result of `rvalue.ty` cannot be used in the type-checker and kind of *may* introduce erased regions, even when all regions that appear in the MIR have been "renumbered". 

What's the motivation exactly for removing this from the MIR? (As opposed to (eventually) encoding a `'erased` here?)

Reply:
The reason for this was that it seemed a bit wasteful to have `Region`s that are almost always erased in MIR (which was my original intention).